### PR TITLE
JSDoc: Remove duplicates in relation-fixer plugin

### LINKED ISSUE #261: JSDoc: Remove duplicates in relation-fixer plugin
Make sure that arrays added to doclets by relation-fixer plugin have no duplicates in them.

### PR DESCRIPTION
**Pull request by m-turek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Remove duplicates in JSDoc relation-fixer plugin. Closes #261 .

---

### Additional information

Related issue: https://github.com/cksource/umberto/issues/228

### CODE REVIEW

File: packages/jsdoc-plugins/tests/relation-fixer/buildrelations.js [hunk 1]
```diff
@@ -94,6 +114,25 @@ describe( 'JSDoc relation-fixer buildrelations module', () => {
...
+			expect( testedDoclet ).to.have.property( 'descendants' ).and.to.include( prop );
+		} );
+	} );
+
+	it( 'should not allow duplicates in relation arrays', () => {
+		const newDoclets = buildRelations( testDoclets );
+		const testedClass = newDoclets.find( d => d.longname === 'classB' );
+		const testedInterface = newDoclets.find( d => d.longname === 'interfaceB' );
+
+		// removing duplicates
```
Comment (Reinmar) [inline packages/jsdoc-plugins/tests/relation-fixer/buildrelations.js]: "Removing duplicates."

File: packages/jsdoc-plugins/tests/relation-fixer/buildrelations.js [hunk 2]
```diff
@@ -94,6 +114,25 @@ describe( 'JSDoc relation-fixer buildrelations module', () => {
...
+		const expectedImplementsNested = Array.from( new Set( testedClass.implementsNested ) );
+		const expectedMixesNested = Array.from( new Set( testedClass.mixesNested ) );
+		const expectedAugmentsNested = Array.from( new Set( testedClass.augmentsNested ) );
+		const expectedDescendants = Array.from( new Set( testedInterface.descendants ) );
+
+		expect( testedClass ).to.have.property( 'implementsNested' ).and.to.eql( expectedImplementsNested );
```
Comment (Reinmar) [inline packages/jsdoc-plugins/tests/relation-fixer/buildrelations.js]: is `eql` the same as `deep.equal()`? If so, I strongly prefer the second option.

File: packages/jsdoc-plugins/lib/relation-fixer/buildrelations.js [hunk 3]
```diff
@@ -39,11 +39,16 @@ function addRelationArrays( originalDoclets ) {
 			relations: [ 'augments', 'implements', 'mixes' ]
 		} );
 
+		// remove duplicates
```
Comment (Reinmar) [inline packages/jsdoc-plugins/lib/relation-fixer/buildrelations.js]: "Remove duplicates."

File: packages/jsdoc-plugins/tests/relation-fixer/buildrelations.js (hunk 2 above)
Comment (m-turek) [inline packages/jsdoc-plugins/tests/relation-fixer/buildrelations.js]: http://chaijs.com/api/bdd/#method_eql:
> The .deep.equal assertion is almost identical to .eql but with one difference: .deep.equal causes deep equality comparisons to also be used for any other assertions that follow in the chain.

I'll change it to `deep.equal`

**Pull request merged**